src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,2068 @@
+        const promiseToGetLatestBlock = blockTracker.getLatestBlock();
+        expect(blockTracker.isRunning()).toBe(true);
+        // We have to wait for the promise to resolve after the assertion
+        // because by the time this promise resolves, the block tracker isn't
+        // running anymore
+        await promiseToGetLatestBlock;
+      });
+    });
+
+    it('should stop the block tracker automatically after its promise is fulfilled', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(async ({ blockTracker }) => {
+        await blockTracker.getLatestBlock();
+        expect(blockTracker.isRunning()).toBe(false);
+      });
+    });
+
+    it('should fetch the latest block number', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(
+        {
+          provider: {
+            stubs: [
+              {
+                methodName: 'eth_blockNumber',
+                response: {
+                  result: '0x0',
+                },
+              },
+            ],
+          },
+        },
+        async ({ blockTracker }) => {
+          const latestBlockNumber = await blockTracker.getLatestBlock();
+          expect(latestBlockNumber).toStrictEqual('0x0');
+        },
+      );
+    });
+
+    it('should cause the request to get the latest block to be made with `skipCache: true` if setSkipCacheFlag: true is given', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(
+        { blockTracker: { setSkipCacheFlag: true } },
+        async ({ provider, blockTracker }) => {
+          jest.spyOn(provider, 'sendAsync');
+
+          await blockTracker.getLatestBlock();
+
+          expect(provider.sendAsync).toHaveBeenCalledWith(
+            {
+              jsonrpc: '2.0' as const,
+              id: expect.any(Number),
+              method: 'eth_blockNumber' as const,
+              params: [],
+              skipCache: true,
+            },
+            expect.any(Function),
+          );
+        },
+      );
+    });
+
+    it('should not cause the block tracker to make a request for the latest block number more than once when called more than once while the current block number cache is still fresh', async () => {
+      recordCallsToSetTimeout();
+
+      await withPollingBlockTracker(async ({ provider, blockTracker }) => {
+        const sendAsyncSpy = jest.spyOn(provider, 'sendAsync');
+        await blockTracker.getLatestBlock();
+        await blockTracker.getLatestBlock();
+        const requestsForLatestBlock = sendAsyncSpy.mock.calls.filter(
+          (args) => {
+            return args[0].method === 'eth_blockNumber';
+          },
+        );
+        expect(requestsForLatestBlock).toHaveLength(1);
+      });
+    });
+
+    it('should cause the block tracker to make a request for the latest block number again if called again after the current block number cache becomes stale', async () => {
```

Comment:
```suggestion
    it('should ask for a new block number if the cached block number is stale', async () => {
```

Reply:
Addressed in 130aabea3ad7cd9923828525a8e0338be93840d2 and commits/aaa6a794ae8a36813fb843f0ba6761dce26f3ee3.